lunatrace/bsl/backend/src/vulnerabilities/handler.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright by LunaSec (owned by Refinery Labs, Inc)
+ *
+ * Licensed under the Business Source License v1.1
+ * (the "License"); you may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at
+ *
+ * https://github.com/lunasec-io/lunasec/blob/master/licenses/BSL-LunaTrace.txt
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+import os from 'os';
+import path from 'path';
+
+import { Request, Response } from 'express';
+import semver from 'semver';
+
+import { GrypeVuln, Convert as VulnCast } from '../types/grypeVulnerabilities';
+import { GrypeMeta, Convert as MetaCast } from '../types/grypeVulnerabilityMetadata';
+import { DbType, initDb } from '../utils/vulnerability-loader';
+
+let db: DbType | null = null;
+
+void (async () => {
+  const grypeDbPath = path.join(os.homedir(), '.cache/grype/db/3/vulnerability.db');
+  db = await initDb(grypeDbPath);
```

Comment:
I think id probably synchronously return this promise and then await it inside the functions that need to use it.  Those are async functions anyway, and it ends up a bit cleaner than checking if the db exists and failing somewhat silently with []

Reply:
`const db = await dbPromise`

Reply:
an anymous top level async function kind of reeks but I understand how we got here.  It would be nice if we could figure out how to turn top-level await on, I know it IS possible

Reply:
If this is really just throwaway code then nevermind